pkg/logql/log/pipeline.go

```diff
@@ -42,6 +43,7 @@ func NewNoopPipeline() Pipeline {
 
 type noopPipeline struct {
 	cache map[uint64]*noopStreamPipeline
+	mu    sync.RWMutex
```

Comment:
I'm unfamiliar with this part of the code so I'm asking just for clarification, `noopPipeline` is only used by tests, right?

Reply:
```
package syntax

func (e *MatchersExpr) Pipeline() (log.Pipeline, error) {
	return log.NewNoopPipeline(), nil
}
```

MatchersExpr